[fix] Qwant engine - implement API v3 and add engine "qwant videos"

return42 opened the pull request:
## What does this PR do?

Implement Qwant API v3, the old API (https://api.qwant.com/api) seems no longer work.

I guess the offset (aka paging) needs more tests.

return42 committed:
[fix] Qwant engines - implement API v3 and add 'quant videos'

The implementation uses the Qwant API (https://api.qwant.com/v3). The API is
undocumented but can be reverse engineered by reading the network log of
https://www.qwant.com/ queries.

This implementation is used by different qwant engines in the settings.yml::

  - name: qwant
    categories: general
    ...
  - name: qwant news
    categories: news
    ...
  - name: qwant images
    categories: images
    ...
  - name: qwant videos
    categories: videos
    ...

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on searx/engines/qwant.py:
```diff
@@ -25,98 +49,142 @@
         params['url'] += '&locale=' + language.replace('-', '_').lower()
 
-    params['headers']['User-Agent'] = 'Mozilla/5.0 (X11; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0'
+    params['headers']['User-Agent'] = gen_useragent()
```
This is automatically set by https://github.com/searxng/searxng/blob/ff180cf2dbab35d93b013b68afc19b795e7b9b71/searx/search/processors/online.py#L61

return42 replied:
Thanks for the hint, I fixed it and rebased. I also added the missing data.language for the new engine "qwant videos".

In the second patch I _enabled_ the Qwant engines .. is this OK .. meanings?

PR is now ready to review

return42 committed:
[mod] enable Qwant engines (web, news, images, video) by default

Qwant is a fast and reliable search engine and AFAIK there is no CAPTCHA.  Let
us enable Qwant engines by default.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
